easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -472,47 +472,53 @@ def dump(self, fp):
         """
         eb_file = file(fp, "w")
 
-        def to_str(x):
-            """Return quoted version of x"""
-            if isinstance(x, basestring):
-                if '\n' in x or ('"' in x and "'" in x):
-                    return '"""%s"""' % x
-                elif "'" in x:
-                    return '"%s"' % x
-                else:
-                    return "'%s'" % x
-            else:
-                return "%s" % x
-
         # ordered groups of keys to obtain a nice looking easyconfig file
         grouped_keys = [
+            ['easyblock'],
             ['name', 'version', 'versionprefix', 'versionsuffix'],
             ['homepage', 'description'],
             ['toolchain', 'toolchainopts'],
-            ['source_urls', 'sources'],
+            ['sources', 'source_urls'],
             ['patches'],
             ['builddependencies', 'dependencies', 'hiddendependencies'],
+            ['osdependencies'],
+            ['preconfigopts', 'configopts'],
+            ['prebuildopts', 'buildopts'],
+            ['preinstallopts', 'installopts'],
             ['parallel', 'maxparallel'],
-            ['osdependencies']
         ]
 
+        last_keys = ['sanity_check_paths', 'moduleclass']
+
+        def check_and_print(keyset):
```

Comment:
last remark (for this PR, at least ;-)): rename to `include_defined_parameters`, and change docstring to

```python
"""
Internal function to include parameters in the dumped easyconfig file which have a non-default value.
"""
```